### PR TITLE
Introduce byte_io.h including byte sequence reading utils (ReadValueBE and ReadValueLE)

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

#### Why we need this PR?
Need to unify byte stream parsing code to increase readability and decrease duplication

#### What happened in this PR?
- Introduced two template functions `ReadValueBE` and `ReadValueLE` to read values from a byte stream in big-endian or little-endian
- Adjusted PNG and BMP PeekShaple implementations to use the new utility functions

**JIRA TASK**: [DALI-XXXX]

### CODE REVIEW

**Comment by jantonguirao** [comment]
!build

File: dali/image/bmp.cc [hunk 1]
```diff
@@ -13,29 +13,36 @@
...
+    w = ReadValueLE<uint16_t>(bmp + 18);
+    h = ReadValueLE<uint16_t>(bmp + 20);
     // BITMAPINFOHEADER and later: | 32u header | 32s width | 32s height | ...
   } else if (length >= 26 && header_size >= 40) {
-    w = static_cast<int>(bmp[18] | bmp[19] << 8 | bmp[20] << 16 | bmp[21] << 24);
-    h = abs(static_cast<int>(bmp[22] | bmp[23] << 8 | bmp[24] << 16 | bmp[25] << 24));
+    w = ReadValueLE<int32_t>(bmp + 18);
+    h = abs(ReadValueLE<int32_t>(bmp + 22));
   }
+  std::cout << h << " x " << w << " x " << 0 << std::endl;
```
Comment (mzient) [inline dali/image/bmp.cc]: remove

Comment (jantonguirao) [reply dali/image/bmp.cc]: done

File: dali/image/bmp.cc [hunk 2]
```diff
@@ -13,28 +13,34 @@
...
 
 namespace dali {
 
-BmpImage::BmpImage(const uint8_t *encoded_buffer, size_t length, DALIImageType image_type) :
-        GenericImage(encoded_buffer, length, image_type) {
-}
+// https://en.wikipedia.org/wiki/BMP_file_format#DIB_header_(bitmap_information_header)
+
+namespace {
+  constexpr int kHeaderSizeOffset = 14;
```
Comment (mzient) [inline dali/image/bmp.cc]: Why? Why only this?

**Comment by jantonguirao** [comment]
!build

File: dali/image/bmp.cc (hunk 2 above)
Comment (jantonguirao) [reply dali/image/bmp.cc]: gone

**Pull request merged**